operator/cmd/ccnp_event.go

```diff
@@ -84,8 +85,8 @@ func enableCCNPWatcher(ctx context.Context, wg *sync.WaitGroup, clientset k8sCli
 			},
 			UpdateFunc: func(oldObj, newObj interface{}) {
 				k8sEventMetric(resources.MetricCCNP, resources.MetricUpdate)
-				if oldCNP := k8s.ObjToSlimCNP(oldObj); oldCNP != nil {
-					if newCNP := k8s.ObjToSlimCNP(newObj); newCNP != nil {
+				if oldCNP := k8s.ObjToGeneric[types.SlimCNP](oldObj); oldCNP != nil {
```

Comment:
Suggestion: While you're in here, add a `InformerObjDeepEqual[]` function too? That should cut down on even more boilerplate :-)

Reply:
`DeepEqual`s are generated functions. I will leave it for another time, but would you prefer to implement `DeepEqual` in a generic way or have a call to `DeepEqual` in a generic function thus saving a few lines of code here and there?